### ISSUE TITLE
Proposal: do not create a directory in tmp when the DefaultOptions variable is initialized

### ISSUE THREAD

**Issue by lapter57** [rosedblabs/rosedb#319]
Hi, can you tell me why when initializing default options a directory is created in tmp. This directory is created in any case, even when the default options are not used or the DirPath field is overridden. At the moment every time I run an application that uses rosedb, an empty directory in tmp is created and I have to delete it. I would suggest that instead of using os.MkdirTemp use os.TempDir to create the path to the temporary directory, because if you use Open when there is no directory it will still be created, so I don't see why a directory should be created when the DefaultOptions variable is initialized.

**Comment by roseduan** [comment]
You can specify a dir path when open the db.

**Comment by lapter57** [comment]
Yes I can do it, but as I wrote above the directory in tmp will be created in any case, because the initialization of the package has already happened, so the DefaultOptions variable is initialized and the tempDBDir function was called where the directory in tmp is created.

**Comment by lapter57** [comment]
create a main.go file:
```go
package main

import (
	"github.com/rosedblabs/rosedb/v2"
)

func main() {
	options := rosedb.DefaultOptions
	options.DirPath = "db_basic"
}
```

and run the script:
```bash
#!/bin/bash

go build main.go

rm -rf $TMPDIR/rosedb-temp*

rosedbDirCnt=$(ls -l $TMPDIR | grep 'rosedb-temp' | wc -l | xargs)

echo "rosedb dirs in tmp is ${rosedbDirCnt}"

rosedbDirCntExpected=100
for i in $(seq ${rosedbDirCntExpected})
do
    ./main
done

rosedbDirCntActual=$(ls -l $TMPDIR | grep 'rosedb-temp' | wc -l | xargs)
echo "rosedb dirs in tmp expected = ${rosedbDirCntExpected} and actual = ${rosedbDirCntActual}"
```
 you will see that the number of directories in tmp will be equal to 100

**Comment by roseduan** [comment]
Oh, I see, this is indeed a problem.

I plan to fix it like this:
```go
func DefaultOptions(path string) Options {
	if len(path) == 0 {
		path = tempDBDir()
	}
	return Options{
		DirPath:           path,
		SegmentSize:       1 * GB,
		Sync:              false,
		BytesPerSync:      0,
		WatchQueueSize:    0,
		AutoMergeCronExpr: "",
	}
}
```

Make the `DefaultOptions` a function, and pass the path as an argument, I will only use the temp dir if the argument is null.
Does this will solve your problem?

**Comment by roseduan** [comment]
Oh I see, it is indeed a problem.

I will fix it.

**Comment by roseduan** [comment]
Just return a temp dir name instead of creating it.
```go
func tempDBDir() string {	
    return filepath.Join(os.TempDir(), "rosedb-temp"+strconv.Itoa(int(nameRand.Int63())))
}
```

Does this slove your problem?

**Comment by lapter57** [comment]
Yes, that will solve the problem

**Comment by roseduan** [comment]
I fixed this in v2.3.9.

thanks for your feedback.
